### ISSUE TITLE
SBT: Unresolved dependencies

### ISSUE THREAD

**Issue by tejasmanohar** [twitter/scrooge#262]
I think the apache thrift dependency version information is out-of-date. I'm getting this error when copying the example SBT config. New to Scala so sorry if I'm missing something silly here.

### Expected behavior

From https://twitter.github.io/scrooge/SBTPlugin.html,
```
libraryDependencies ++= Seq(
  "org.apache.thrift" % "libthrift" % "0.9.2",
  "com.twitter" %% "scrooge-core" % "4.6.0" exclude("com.twitter", "libthrift"),
  "com.twitter" %% "finagle-thrift" % "6.34.0" exclude("com.twitter", "libthrift")
)
```

I'd expect SBT to resolve this.

### Actual behavior

```
Error:Unresolved dependencies:<ul><li> org.apache.thrift#libthrift;0.5.0-1: not found
```

I'm confused where 0.5.0-1 is coming from TBH.

**Comment by tejasmanohar** [comment]
https://github.com/twitter/finagle/pull/160 seems relevant, but it's "closed". :\

**Comment by bryce-anderson** [comment]
This blog post may help you: https://finagle.github.io/blog/2016/11/29/central-libthrift/

TL;DR: you're using an older version of finagle-thrift that depends on that older version of libthrift which is not published to maven central, but is published to our own maven repository. You could add a resolver, or better yet, update to a newer version of finagle-thrift and do the manual exclusion since you're shadowing it with a newer version of libthrift.

**Comment by tejasmanohar** [comment]
Ahhh thanks! I'll try that

**Comment by tejasmanohar** [comment]
compilation succeeds with

build.sbt
```scala
libraryDependencies ++= Seq(
  "com.twitter" %% "finagle-thrift" % "6.42.0",
  "org.apache.thrift" % "libthrift" % "0.10.0"
)
```

plugins.sbt
```
addSbtPlugin("com.twitter" % "scrooge-sbt-plugin" % "4.5.0" exclude("org.apache", "libthrift"))
```

is that what you meant, @bryce-anderson? thanks!

**Comment by tejasmanohar** [comment]
I can actually build with no problem until I add the SBT plugin... where neither exclusion nor adding a resolver seems to do the trick :\

_plugins.sbt_
```scala
logLevel := Level.Warn

addSbtPlugin("com.twitter" % "scrooge-sbt-plugin" % "4.5.0" exclude("com.twitter", "libthrift") exclude("org.apache.thrift", "libthrift"))
```

_build.sbt_
```
name := "athens"

version := "1.0"

scalaVersion := "2.12.1"

resolvers += "twitter" at "http://maven.twttr.com"

libraryDependencies ++= Seq(
  "com.twitter" % "scrooge-core_2.12" % "4.14.0" exclude("com.twitter", "libthrift") exclude("org.apache.thrift", "libthrift"),
  "com.twitter" % "finagle-thrift_2.12" % "6.42.0" exclude("com.twitter", "libthrift") exclude("org.apache.thrift", "libthrift")
)
```

Fails with
```
[info] Resolving io.netty#netty-common;4.1.8.Final ...
[info] Resolving io.netty#netty-codec;4.1.8.Final ...
[info] Resolving com.twitter#finagle-netty4_2.12;6.42.0 ...
[info] Resolving io.netty#netty-codec-socks;4.1.8.Final ...
[info] Resolving io.netty#netty-buffer;4.1.8.Final ...
[info] Resolving org.scala-lang.modules#scala-parser-combinators_2.12;1.0.4 ...
[info] Resolving com.twitter#util-stats_2.12;6.41.0 ...
[info] Resolving com.twitter#util-function_2.12;6.41.0 ...
[info] Resolving com.fasterxml.jackson.core#jackson-core;2.8.4 ...
[info] Resolving com.fasterxml.jackson.module#jackson-module-scala_2.12;2.8.4 ...
[info] Resolving io.netty#netty-resolver;4.1.8.Final ...
[info] Resolving org.scala-lang#scala-library;2.12.1 ...
[info] Resolving com.google.guava#guava;16.0.1 ...
[info] Resolving org.scala-lang#scala-compiler;2.12.1 ...
[info] Resolving org.scala-lang.modules#scala-xml_2.12;1.0.6 ...
[info] Resolving jline#jline;2.14.1 ...
[trace] Stack trace suppressed: run 'last *:updateSbtClassifiers' for the full output.
[error] (*:updateSbtClassifiers) sbt.ResolveException: unresolved dependency: org.apache.thrift#libthrift;0.5.0-1: not found
```

**Comment by mosesn** [comment]
@tejasmanohar you need to use the same version of the scrooge plugin as the scrooge you're using.  up to date versions of the scrooge plugin will com.twitter libthrift which is published to maven central instead of org.apache.thrift 0.5.0-1, which is not.

**Comment by tejasmanohar** [comment]
Ahh, I'll try that

**Comment by sihil** [comment]
I've also just run into this. I think partly because the versions on the docs at https://twitter.github.io/scrooge/SBTPlugin.html are a year or so out of date.
